src/tracer.js

```diff
@@ -0,0 +1,130 @@
+'use strict'
+
+import BasicSpan from './span'
+import {TextMapPropagator, BinaryPropagator} from './propagation'
+import Sampler from './sampler'
+import Recorder from './recorder'
+
+export default class BasicTracer {
+    constructor({sampler, recorder} = {}) {
+        this._sampler = sampler || new Sampler()
+        this._recorder = recorder || new Recorder()
+        this._binaryPropagator = new BinaryPropagator(this)
+        this._textPropagator = new TextMapPropagator(this)
+    }
+    setInterface(inf) {
+        this._interface = inf
+    }
+    /**
+     * Starts and returns a new Span representing a logical unit of work.
```

Comment:
nit: no need for the repeats of OpenTracing comments... it's probably better to just include a pointer in a `BasicTracer` class comment to the OpenTracing-js Tracer spec.